client/state/plugins/recommended/reducer.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import { keyedReducer } from 'state/utils';
+import {
+	PLUGINS_RECOMMENDED_RECEIVE,
+	PLUGINS_RECOMMENDED_REQUEST_FAILURE,
+} from 'state/action-types';
+
+const reducer = keyedReducer( 'siteId', ( state = null, action ) => {
+	switch ( action.type ) {
+		case PLUGINS_RECOMMENDED_RECEIVE: {
+			return action.data;
+		}
+		case PLUGINS_RECOMMENDED_REQUEST_FAILURE: {
```

Comment:
I'm not sure about this one: if we already have the recommended plugins from a previous request, and a new request is made, but it fails, wouldn't it be better if we just keep the previous data instead of wiping it? I'd expect we don't really do anything with the existing data if the request fails.

Perhaps we have a good reason to do this?

Reply:
My original intent was to ensure that an empty interface would be shown if the recommendations request fails. Let me figure a way to handle that scenario without having to wipe out the previous data...